calico_containers/tests/st/workload.py

```diff
@@ -0,0 +1,100 @@
+from netaddr import IPAddress, AddrFormatError
+
+
+class Workload(object):
+    """
+    A calico workload.
+
+    These are the end-users containers that will run application-level software.
+    """
+    def __init__(self, host, name, ip=None, image="busybox", use_powerstrip=True):
+        """
+        Create the workload and detect its IPs.
+
+        :param host: The host container on which this workload is instantiated.
+        All commands executed by this container will be passed through the host
+        via docker exec.
+        :param name: The name given to the workload container. This name is
+        passed to docker and can be used inside docker commands.
+        :param ip: The IP to be assigned to this workload via calico. May be
```

Comment:
It would be really helpful to be more explicit about what is allowed for this parameter.  I think it's "auto", None, or a string representing the IP address.